gtfs-realtime/spec/en/reference.md

```diff
@@ -554,13 +561,132 @@ A localized image URL mapped to a language.
 
 ## _message_ Shape
 
-Describes the physical path that a vehicle takes when the shape is not part of the (CSV) GTFS, such as for an ad-hoc detour. Shapes belong to Trips and consist of an encoded polyline for more efficient transmission.  Shapes do not need to intercept the location of Stops exactly, but all Stops on a trip should lie within a small distance of the shape for that trip, i.e. close to straight line segments connecting the shape points
+Describes the physical path that a vehicle takes when the shape is not part of the (CSV) GTFS, such as for an ad-hoc detour. Shapes belong to Trips and consist of an encoded polyline for more efficient transmission.  Shapes do not need to intercept the location of Stops exactly, but all Stops on a trip should lie within a small distance of the shape for that trip, i.e. close to straight line segments connecting the shape points.
 
-**Caution:** this message is still **experimental**, and subject to change. It may be formally adopted in the future.<br>.
+<br><br>**Caution:** this message is still **experimental**, and subject to change. It may be formally adopted in the future.
 
 **Fields**
 
 | _**Field Name**_ | _**Type**_ | _**Required**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|----------------|-------------------|-------------------|
 | **shape_id** | [string](https://protobuf.dev/programming-guides/proto2/#scalar) | Required | One |  Identifier of the shape. Must be different than any `shape_id` defined in the (CSV) GTFS. <br><br>**Caution:** this field is still **experimental**, and subject to change. It may be formally adopted in the future. |
 | **encoded_polyline** | [string](https://protobuf.dev/programming-guides/proto2/#scalar) | Required | One | Encoded polyline representation of the shape. This polyline must contain at least two points. For more information about encoded polylines, see https://developers.google.com/maps/documentation/utilities/polylinealgorithm <br><br>**Caution:** this field is still **experimental**, and subject to change. It may be formally adopted in the future. |
+
+## _message_ Stop
+
+Represents a new Stop added to the feed dynamically. All fields are as described in the (CSV) GTFS specification. The location type of the new stop is `0` (routable stop). 
+
+<br><br>**Caution:** this field is still **experimental**, and subject to change. It may be formally adopted in the future.
+
+**Fields**
+
+| _**Field Name**_ | _**Type**_ | _**Required**_ | _**Cardinality**_ | _**Description**_ |
+|------------------|------------|----------------|-------------------|-------------------|
+| **stop_id** | [string](https://protobuf.dev/programming-guides/proto2/#scalar) | Required | One |  Identifier of the stop. Must be different than any `stop_id` defined in the (CSV) GTFS. |
```

Comment:
In the proto file this field is optional. I know there was something with optional fields being recommend in protobuf but I don't remember the details.

All I know that we have a similar situation with alters where header texts are required by the spec text but not the proto and I regularly have to discuss with people if they are mandatory or not.

Reply:
https://protobuf.dev/programming-guides/dos-donts/#add-required
https://news.ycombinator.com/item?id=18196288

Generally speaking, new "required" fields in protobufs are discouraged.  It's less about high-level goals of data validation and more about what happens when different data producers and consumers (and intermediates) with slightly different specs are involved + nuances of the protobuf implementation itself.

I actually thought about flagging this originally, since I was trying to think of a case where we might not specify a `stop_id` as part of `Stop` at some point down the road (maybe it's possible?).  But I would not add more required fields unnecessarily.

Reply:
Yes, I agree this is a bit strange but it's the way to go to avoid problem or breaking changes in the future. Part of the problem is the documentation is in the .proto too so it's used standalone. We might want to add requirement of the field in documentation in the .proto in addition to the reference.